server/gql/models/aggregate/purchasedProductsUtils.js

```diff
@@ -28,3 +30,46 @@ export const queryOptions = args => ({
   },
   type: QueryTypes.SELECT
 });
+
+export const queryRedis = async (type, args) => {
+  let startDate;
+  let endDate;
+  let count = 0;
+  if (!args?.startDate) {
+    const createdAtDates = await getEarliestCreatedDate();
+    startDate = createdAtDates.toISOString().split('T')[0];
+  } else {
+    startDate = args.startDate.toISOString().split('T')[0];
+  }
+  if (!args?.endDate) {
+    endDate = moment().format('YYYY-MM-DD');
+  } else {
+    endDate = args.endDate.toISOString().split('T')[0];
+  }
+  if (args?.category) {
+    do {
+      const totalForCategory = await redis.get(`${startDate}_${args.category}`);
+      if (totalForCategory) {
+        const parsedTotalForCategory = JSON.parse(totalForCategory);
```

Comment:
this may throw an error. Let's please handle gracefully

Reply:
the while seems repatative. Regardless of the condition we run the same while loop. The only thing that changes is the redis key. Let's change the value of the key based on whether args.category is present